### PR TITLE
Use 'greedy' skipEmptyLines in unparse

### LINKED ISSUE #553: Either skipEmptyLines is broken--or is not used for unparse()
**Expected**
Calling unparse with skipEmptyLines: 'greedy' results in no basically empty rows.

**Actual**
Calling unparse with skipEmptyLines: 'greedy' (or true) will still have rows with no data (except commas). 

**Sample Code**
```
let csv = Papa.unparse(
      {
        fields: ['a', 'b'],
        data: [[null, ' '], [], ['1', '2']]
      },
      {
        skipEmptyLines: 'greedy',
        worker: true
      }
 );
console.log(csv);
```

**Discussion**
Am I misunderstanding something or not setting the config correctly? Is `skipEmptyLines` only for parse, perhaps? In the mean time, I suppose I can do some edge checking so I don't pass "empty" rows to unparse.... :(

Keep up the good work! I love the project. Glad I found it. Fast, clean, modern, reliable.

### PR DESCRIPTION
**Pull request by MonkeyDZeke**
Added support for skipEmptyLines in unparsing using both `true` and `'greedy'` values.

Resolves #553

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -335,6 +338,10 @@
 				|| _config.quotes instanceof Array)
 				_quotes = _config.quotes;
 
+			if (typeof _config.skipEmptyLines === 'boolean'
```
Comment (pokoli) [inline papaparse.js]: Why testing the typeof skipemptyLines? Is his really required?

**Comment by MonkeyDZeke** [comment]
Setting a default value for it is required, or else testing it will throw
an error if it's not set. I followed the coding pattern for that already
established, which includes checking the data type when overriding the
defaults with the config.

**Comment by pokoli** [comment]
@MonkeyDZeke that makes sense. Thanks for the explanation!

**Comment by MonkeyDZeke** [comment]
You're welcome! :)

File: tests/test-cases.js [hunk 2]
```diff
@@ -1636,6 +1636,24 @@ var UNPARSE_TESTS = [
...
+		expected: '," "\r\n\r\n1,2'
+	},
+	{
+		description: "Returns without empty rows when skipEmptyLines is true",
+		input: [[null, ' '], [], ['1', '2']],
+		config: {skipEmptyLines: true},
+		expected: '," "\r\n1,2'
+	},
+	{
+		description: "Returns without rows with no content when skipEmptyLines is 'greedy'",
```
Comment (bean5) [inline tests/test-cases.js]: :+1:

**Pull request merged**